pkg/route/hostassignment/externalcertificate_test.go

```diff
@@ -0,0 +1,209 @@
+package hostassignment
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routecommon "github.com/openshift/library-go/pkg/route"
+)
+
+func TestValidateHostExternalCertificate(t *testing.T) {
+
+	type args struct {
```

Comment:
I don't see a reason for a dedicated structure for the function arguments. The idiomatic go table tests pass args as fields of `testCase` struct.

Reply:
I think the thought process could be to make the testCases look a bit more structured, but I agree that go table tests pass args as fields of testCase struct. I've updated them, thanks. 
I've also added a few more tests to cover other scenarios.